compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -1033,6 +1033,9 @@ private List<BIROperand> generatePositionArgs(BIRPackage pkg, BIRFunction func,
                                                   Location location) {
         String pkgId = generatePackageId(pkg.packageID);
         BIROperand pkgOperand = generateGlobalConstantOperand(pkg, symbolTable.stringType, pkgId);
+        if (location == null) {
+            location = new BLangDiagnosticLocation("", 0, 0, 0, 0, 0, 0);
+        }
```

Comment:
I don't think this is the correct fix. Why is location null in the first place?

Reply:
The location here is taken from the position of terminator in the BIR block which is null in the case of the anonymous function for getting Foo.bar which causes the problem. Should it have a position, if so should we inject that in BIRGen?

Reply:
We shouldn't be updating the pos in the backend. If at all, it should have been set properly in the frontend.

Reply:
the BlangFucntionBody/ statements having null position is not incorrect behavior in this case right? Or should they inherit some position?

Reply:
What is the pos that becomes null here? Is it that of `foo.bar` in 

```ballerina
service class Foo {
    remote function bar() {
        
    }
}

public function main() {
    Foo foo = new;
    any x = foo.bar;
}
```

Did we check how the pos is set for a (non-method) function reference?

Reply:
Yes, the anonymous function to get `foo.bar` gets null pos